web3/main.py

```diff
@@ -130,8 +136,12 @@ def __init__(self, provider=None, middlewares=None, modules=None, ens=empty):
         if modules is None:
             modules = get_default_modules()
 
-        for module_name, module_class in modules.items():
-            module_class.attach(self, module_name)
+        for module in modules:
+            module_class, *submodules = modules[module]
+            module_class.attach(self, module)
+            if submodules:
+                for subname, subclasses in submodules[0].items():
+                    subclasses[0].attach(getattr(self, module), subname)
 
```

Comment:
@carver Thoughts on this setup? Maybe there's a cleaner way to achieve the attaching of subclasses?